reconcile/slack_usergroups.py

```diff
@@ -277,14 +277,20 @@ def get_desired_state(slack_map):
             channel_names = [] if p['channels'] is None else p['channels']
             channels = slack.get_channels_by_names(channel_names)
 
-            desired_state.append({
-                "workspace": workspace_name,
-                "usergroup": usergroup,
-                "usergroup_id": ugid,
-                "users": users,
-                "channels": channels,
-                "description": description,
-            })
+            existing_items = [i for i in desired_state
+                              if i['workspace'] == workspace_name
+                              and i['usergroup'] == usergroup]
+            if len(existing_items) == 1:
+                existing_items[0]['users'].update(users)
```

Comment:
Is there a possibility that the already existing `users` is different to the new one? I'm trying to find the answer for myself but I guess you want a faster review 😅

Reply:
the existing users are likely different. if user files reference roles and roles reference a slack usergroup, different users reference different roles which may reference the same slack usergroup. this is what we are solving.

Reply:
As long as we're ok with the last users list winning... :-D

Reply:
but why?
users is a dict, and `update` adds items to the dict...

Reply:
because I hadn't properly read the code 😂 😅

Reply:
that's a good reason! thanks a lot!